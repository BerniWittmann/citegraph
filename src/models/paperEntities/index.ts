import Author from './author'
import Conference from './conference'
import Country from './country'
import Institution from './institution'
import Publication from './publication'
import Publisher from './publisher'
import Record from './record'
import PaperEntity, { PaperEntityTableColumn, PaperEntityFields } from './base'

export const entities = [
  Record,
  Author,
  Institution,
  Country,
  Publication,
  Publisher,
  Conference
]

export const entityKeysMap = {
  [Record.key]: Record,
  [Author.key]: Author,
  [Institution.key]: Institution,
  [Country.key]: Country,
  [Publication.key]: Publication,
  [Publisher.key]: Publisher,
  [Conference.key]: Conference
}

export const entityKeys: Array<string> = entities.map((entity) => entity.key)

export type PaperEntityTableColumn = PaperEntityTableColumn

export type PaperEntity = PaperEntity

export type PaperEntityFields = PaperEntityFields

export default {
  Author,
  Conference,
  Country,
  Institution,
  Publication,
  Publisher,
  Record
}
