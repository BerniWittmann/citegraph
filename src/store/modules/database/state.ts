import { DatabaseState } from './types'

export const state: DatabaseState = {}
