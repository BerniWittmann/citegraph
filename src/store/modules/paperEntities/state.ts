import { PaperEntitiesState } from './types'

export const state: PaperEntitiesState = {
  entityType: undefined,
  entities: [],
  entityCount: -1
}
