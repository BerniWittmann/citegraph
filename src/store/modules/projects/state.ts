import { ProjectsState } from './types'

export const state: ProjectsState = {
  activeProject: undefined,
  projects: [],
  openProjects: []
}
