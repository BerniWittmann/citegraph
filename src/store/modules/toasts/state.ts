import { ToastsState } from './types'

export const state: ToastsState = {
  toasts: []
}
