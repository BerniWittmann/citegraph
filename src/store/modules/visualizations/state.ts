import { VisualizationsState } from './types'

export const state: VisualizationsState = {
  visualizations: [],
  currentVisualization: undefined
}
