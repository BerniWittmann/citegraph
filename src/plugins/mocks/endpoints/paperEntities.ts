import db from '../db'
import MockAdapter from 'axios-mock-adapter'
import { RecordFields } from '@/models/paperEntities/record'
import { setupMocks as setupGraphQLMocks } from './graphql-helpers/endpoint'
import { QueryConfiguration } from './graphql-helpers/queries'
import { AuthorFields } from '@/models/paperEntities/author'
import { CountryFields } from '@/models/paperEntities/country'

const records = db.getCollection('records')
const authors = db.getCollection('authors')
const countries = db.getCollection('countries')

const queryResponseConfiguration: Record<string, QueryConfiguration> = {
  'Records': {
    schemaName: 'record',
    queryName: 'Records',
    collection: records,
    filterFunction: recordsFilter
  },
  'Authors': {
    schemaName: 'author',
    queryName: 'Authors',
    collection: authors,
    filterFunction: authorsFilter
  },
  'Countries': {
    schemaName: 'country',
    queryName: 'Countries',
    collection: countries,
    filterFunction: countriesFilter
  }
}

function recordsFilter (obj: RecordFields, filter: string) {
  return obj.title.toLowerCase().includes(filter) ||
    obj.year.toString().includes(filter) ||
    obj.keywords.map(keyword => keyword.toLowerCase()).join('').includes(filter) ||
    obj.authors.map(author => (author.firstName + ' ' + author.lastName).toLowerCase()).join(';').includes(filter)
}

function authorsFilter (obj: AuthorFields, filter: string) {
  return obj.firstName.toLowerCase().includes(filter) ||
    obj.lastName.toLowerCase().includes(filter)
}

function countriesFilter (obj: CountryFields, filter: string) {
  return obj.name.toLowerCase().includes(filter)
}

export function setupMocks (mock: MockAdapter) {
  setupGraphQLMocks(mock, queryResponseConfiguration)
}
