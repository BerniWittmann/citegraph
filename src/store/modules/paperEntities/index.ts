import { Module } from 'vuex'
import { PaperEntitiesState } from './types'
import { RootState } from '@/store/types'
import { state } from './state'
import { actions } from './actions'
import { getters } from './getters'
import { mutations } from './mutations'

const namespaced: boolean = true

export const paperEntities: Module<PaperEntitiesState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
}
