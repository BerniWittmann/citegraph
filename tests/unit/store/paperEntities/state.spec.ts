import { state } from '@/store/modules/paperEntities/state'

describe('store/modules/paperEntities/state', () => {
  it('provides the initial state', () => {
    expect(state).toMatchSnapshot()
  })
})
