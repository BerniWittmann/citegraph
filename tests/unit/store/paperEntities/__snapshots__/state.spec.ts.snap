// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/modules/paperEntities/state provides the initial state 1`] = `
Object {
  "entities": Array [],
  "entityCount": -1,
  "entityType": undefined,
}
`;
